telemetry/telemetry.proto

```diff
@@ -4,6 +4,57 @@ import 'nanopb.proto';
 
 option java_outer_classname="TelemetryProto";
 
-message Telemetry {
-  optional int32 timestamp_ms = 1;
+//Should be sent only after device resets
+message NameIDMessage { //Send the device ID with its corresponding name
+	optional uint32 id = 1;
+	optional string name = 2 [(nanopb).max_size = 32];
+}
+
+message StatisticalAggregate {
+  optional uint32 numSamples = 1;
+  optional uint32 min = 2;
+  optional uint32 max = 3;
+  optional uint32 mean = 4;
+  optional uint32 stdev = 5;
+}
+
+message GeneralReading { //this might want a more specific name?
+	enum READING_TYPE {
+		VOLTAGE = 1;
+		CURRENT = 2;
+		TEMPERATURE = 3;
+	}
+	optional READING_TYPE type = 1;
+	optional uint32 id = 2; //id of the device being read
+	optional StatisticalAggregate reading = 3;
+}
+
+message MCCInfoMessage {
+	optional StatisticalAggregate speed = 1; //information about the entire car's speed
+}
+
+message PedalMessage {
+	optional StatisticalAggregate accelPos = 1;
```

Comment:
Telemetry can re-format data, you can unpack two uint16's (or however it's packed, assuming it is) from the accelerator/brake system CAN message and put them into two measurement aggregates to be sent out as two different Measurement messages.

In general, the idea here is to send high-level data (voltages, user interface positions, etc) instead of low-level CAN messages, and Telemetry will have to do some translation and aggregation with some understanding of the data format.